### ISSUE TITLE
Unable to lookup `PageStorage` when it exists above the `Navigator`

### ISSUE THREAD

**Issue by esDotDev** [flutter/flutter#92873]
I would like to have a single `PageStorage`, above my `MaterialApp` or `Navigator`, that can act as a global store for views.

For some reason, if I place my `PageStorage` widget outside of a page route, then I can not access it.

```
final bucket = PageStorageBucket();
class PageStorageIssue extends StatefulWidget {
  @override
  State<PageStorageIssue> createState() => _PageStorageIssueState();
}

class _PageStorageIssueState extends State<PageStorageIssue> {
  @override
  void initState() {
    super.initState();
    bucket.writeState(context, 'hello', identifier: 'foo');
  }

  @override
  Widget build(BuildContext context) => PageStorage( // I can never reach this widget
        bucket: bucket,
        child: MaterialApp(
          home: HomePage(),
        ),
      );
}

class HomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    // this prints null
    print(PageStorage.of(context)?.readState(context, identifier: 'foo'));
    return PageStorage( // I need to add PageStorage to here in order to find my bucket
        bucket: bucket,
        child: Builder(builder: (context) {
          // this correctly find the bucket
          print(PageStorage.of(context)?.readState(context, identifier: 'foo'));
          return Placeholder();
        }));
  }
}
```

This is a pain because I need to add a `PageStorage` widget to every single top level page in the app, rather than just being able to lookup to one shared parent.

**Comment by darshankawar** [comment]
> there is a PageStorage element on the Route,

That's correct, usually `PageStorage` is already included in the routes and hence could be conflicting with it's explicit usage along with navigator.

Keeping it open for further insights on a solution / workaround on this.

**Comment by esDotDev** [comment]
Would it be possible to give us a hook to disable this baked in behavior? Maybe we could change this to 
_Add support for  `bool MaterialApp.enablePageStorage`?_

I'm not sure what the routes are really doing with this, since they lose state anyways when a route is removed from stage. I think this is so that sub-widgets of a route will retain their state when added/removed from the tree.

**Comment by HerrNiklasRaab** [comment]
@darshankawar Stumbled into this as well...

My use case is, that I have a list view, each listview item in that listview opens a screen when tapped. On this screen, there is a text field. If the user taps on let's say ListView item 1, enteres "asdf" into the textbox navigates back taps on ListView item 2, entered "fdsa", goes back again taps on ListView item 1 again, it should still have "asdf" in the text box. 

What would be the Flutter idiomatic way of doing this? Using bloc for this seems like overkill.

**Comment by HerrNiklasRaab** [comment]
Especially the [documentation](https://api.flutter.dev/flutter/widgets/PageStorageBucket-class.html#:~:text=Useful%20for%20storing%20per%2Dpage%20state%20that%20persists%20across%20navigations%20from%20one%20page%20to%20another.) is confusing if this is not possible:

> Useful for storing per-page state that persists across navigations from one page to another.
